internal/app/components/roles/dave.go

```diff
@@ -2,15 +2,20 @@ package roles
 
 import "github.com/gzcharleszhang/course-planner/internal/app/components/permissions"
 
+const DaveDisplayName = "admin"
```

Comment:
I think it's pretty clear what each role can do just from the constructor, since we are specifying the permissions each role can have

Reply:
Yeah I mean fair it does direct you to the permissions it can have. 

I think it would be nice to have a little description about what it would entail like `Members can view and add timelines, as well as view data...` or `SuperAdmins can change courses offerings and graduation requirements...` since the general `permissions.go` files doesn't tell us much about what Snew can do that Conrad can't do. It helps everyone know for certain what each role entails without digging up the permission from the code.

Reply:
honestly i'm not sure what kind of services we will have, we probably only need `Conrad` for most of the services for now, we can document more when we the roles are more defined

Reply:
I think to better address your question, our current roles and permissions system doesn't support the idea of `Conrad can view or add timelines` or `Dave can delete user accounts`. Currently, roles have permissions, then each route is associated with a permission level e.g. `Dave has access to unauthenticated, authenticated, and admin routes` where `delete user user accounts is an admin route` and `view or add timelines is an authenticated route`. You did bring up something that I'm working towards, which is granular permissions, and that is exactly what you mentioned. Instead of each role having permission levels, each permission will basically be the service/handler name. Since I don't see a point in implementing granular permissions for our project yet, so this could be a future improvement.

Reply:
That makes sense. I guess I did overlook the fact that they were specific to routes and that's the intended function currently. We'll see how the roles and permissions develop I suppose. Looks good for now.